build: upgrade compile docker & support `CONFIG CLAUSE`

aceforeverd opened the pull request:
- upgrade zetasql to v0.2.6
- upgrade compile image to `hybridsql:0.4.1`
- emit 'make thirdparty' from workflow
- fix the java sdk deploy #897
- suuport `CONFIG CLAUSE`: #907

aceforeverd committed:
build: upgrade zetasql to v0.2.6

aceforeverd committed:
build: upgrade to hybrdsql:0.4.1

- emit 'make thirdparty' from workflow

aceforeverd committed:
fix start/stop zookeeper

zhanghaohit reviewed (changes requested):
You may want to replace all the occurrences of `hybridsql:0.4.0` to `hybridsql:0.4.1`, e.g., docs/en/compile.md, hybridse/README.md, and all the others.

aceforeverd committed:
fix cicd workflow

dl239 commented on steps/init_env.sh:
```diff
@@ -17,6 +17,8 @@
 # init_env.sh
 set -eE -x
 
+echo -e "${RED}this script is deprecated, please run 'make thirdparty' in top directory instead${NC}"
```
delete this unused file directly?

aceforeverd replied:
https://github.com/4paradigm/OpenMLDB/blob/main/.github/workflows/integration-test-src.yml not upgraded @wuyou10206

aceforeverd committed:
fix python test

aceforeverd committed:
support Config Clause in logical plan

aceforeverd committed:
docs: update compile.md

upgrade hybridsql to 0.4.1

aceforeverd committed:
resolve `Print()` function in PlanNode relate to config clause

aceforeverd committed:
optimize a bit shared_ptr

aceforeverd committed:
disable union query for physical plan

aceforeverd committed:
refactor(cicd): java sdk jobs

aceforeverd committed:
ci: turn off compiling hybridse test & example code in cicd

aceforeverd committed:
ci: update workflow

aceforeverd commented on hybridse/src/vm/transform.cc:
```diff
@@ -795,6 +795,9 @@ Status BatchModeTransformer::TransformUnionOp(const node::UnionPlanNode* node,
     CHECK_STATUS(
         CreateOp<PhysicalUnionNode>(&union_op, left, right, node->is_all));
+    if (node->config_options_ != nullptr) {
+        union_op->config_options_ = node->config_options_;
```
union query not support

aceforeverd replied:
conflicts with #796, will rm it

aceforeverd committed:
resolve java sdk macos variant

aceforeverd committed:
fix java sdk deploy on mac job

aceforeverd committed:
fix maven deploy

aceforeverd committed:
try again

aceforeverd committed:
refactor: pom.xmls

aceforeverd committed:
write comments for tricky codes

aceforeverd committed:
rm out-of-date/duplicated files

aceforeverd committed:
fix deploy workflow

aceforeverd committed:
ci: add a step to upload failed zookeeper logs if any test fails

aceforeverd committed:
build(pom.xml): update deps

- surefire -> 3.0.0-M5
  - rm deprecated forkMode option
- maven-jar-plugin -> 3.2.0

aceforeverd committed:
docs(java sdk): update dependency management

tobegit3hub reviewed (approved):
LGTM

aceforeverd committed:
ci: rm upload zookeeper log due to not works

removed because it not work as expect.
better solution: there are 4 type of test runs, each restarts zookeeper,
should change to start zookeeper once

aceforeverd committed:
Merge remote-tracking branch 'upstream/main' into build/upgrade-compile-docker
